repos/fdbt-site/src/pages/api/defineCaps.ts

```diff
@@ -0,0 +1,93 @@
+import { NextApiResponse } from 'next';
+import { FULL_CAPS_ATTRIBUTE, CAPS_DEFINITION_ATTRIBUTE } from '../../constants/attributes';
+import { getCapByNocAndId } from '../../data/auroradb';
+import { getAndValidateNoc, redirectTo, redirectToError } from '../../utils/apiUtils/index';
+import { CapInfo, CapSelection, CapsDefinitionWithErrors, NextApiRequestWithSession, PremadeCap } from 'src/interfaces';
+import { updateSessionAttribute } from '../../../src/utils/sessions';
+
+const getCapContent = async (
+    cap: number | undefined,
+    noc: string,
+): Promise<
+    | {
+          dbCap: PremadeCap;
+          caps: CapInfo;
+      }
+    | undefined
+> => {
+    if (!cap) {
+        return undefined;
+    }
+
+    const result = await getCapByNocAndId(noc, cap);
+
+    if (!result) {
+        throw new Error('No premade caps saved under the provided name');
+    }
+
+    const dbCap = { id: result.id, contents: result } as PremadeCap;
+    const caps = result;
+
+    return {
+        dbCap,
+        caps,
+    };
+};
+
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        const { capChoice, cap } = req.body;
+
+        if (!capChoice) {
+            const capDefinitionWithErrors: CapsDefinitionWithErrors = {
+                id: cap,
```

Comment:
If they havent selected yes or no, do we need to return the id? We don't know what has been passed in the request, so i think its better if we dont.

Reply:
As above, if they havent selected yes or no then there is no need to pass back the id - as it is probably not there.